hedyorg/hedy
# 💻 Enhance customize-adventure page

Fixes #4632 

**What?**
There are two main features introduced in this PR. 
- show all classes that use the current adventure in one or more levels.
- replace the plain text editor by a WYSIWYG editor CKEditor.

**How to test?**
go to the customize-adventure page and use the editor and see the classes (if any).

**Notes**
The editor is the main change, thus. See the following: 

![image](https://github.com/hedyorg/hedy/assets/20051470/f6c5734a-b5c1-4291-b9d1-ef38051c8d40)

As can be seen, the toolbar has different plugins that teachers can utilize in order to enhance their experience with writing/creating custom adventures. For instance, teachers can make text bold by any of the obvious button in the toolbar. Alternatively, they could press CTRL-B to do so. Similarly, they can add ordered and unordered lists. Or create different headings for better visualization and attracting students' attention. 


Perhaps we could add a small paragraph that explains such features. But we certainly need to explain that the `code` button makes a keyword, and the code block button creates a pre tag that would be rendered as a code that students can use. The pre tags aren't highlighted still, so further investigation could be done on how to do so. However, i don't think it's really needed since the editor really shows that it's a piece of code. And as Felienne noted that teachers can always preview what they have written so far, with syntax highlighting since that functionality is just HTML and would use codemirror.
